x-pack/legacy/plugins/infra/public/pages/logs/log_entry_rate/use_log_entry_rate_setup.tsx

```diff
@@ -0,0 +1,54 @@
+/*
```

Comment:
I wonder if we can go even further in sharing components and remove this file, and `use_log_rate_module.tsx`

I might be missing something, but this file is almost the same as `use_log_entry_rate_setup`, right? The only difference is that the file from this PR uses the context from `use_log_entry_rate_module`.

I checked [that file](https://github.com/weltenwort/kibana/blob/eaf313d896ad083b57a104bfe45da8f33fd585c3/x-pack/legacy/plugins/infra/public/pages/logs/log_entry_categories/use_log_entry_categories_module.tsx) and it's also the same as its [entry rate counterpart](https://github.com/weltenwort/kibana/blob/eaf313d896ad083b57a104bfe45da8f33fd585c3/x-pack/legacy/plugins/infra/public/pages/logs/log_entry_categories/use_log_entry_categories_module.tsx).

I think we can merge those two files together, and merge the `*setup`  files as well.

Reply:
The idea is that...

* `useLogEntryRateModule` composes the log entry rate module definition and the generic `useLogAnalysis`
* `useLogEntryCategoriesModule` composes the log entry rate module definition and the generic `useLogAnalysis`
* `useLogEntryRateSetup` composes `useLogEntryRateModule` and the generic `useAnalysisSetupState`
* `useLogEntryCategoriesSetup` composes `useLogEntryCategoriesModule` and the generic `useAnalysisSetupState`

So indeed, the differences between the log entry rate and log entry cateogries files are small because they follow the same pattern, but at some point the composition of the module-specific code and the generic code has to take place. Where else would you propose that should happen?

Reply:
> Where else would you propose that should happen?

To be honest I don't have a good answer yet. I would need to dive deeper into the code.

I think it's still good to go ahead now. We can revisit this later